workload_7_postprocess_fn/main_test.py

```diff
@@ -0,0 +1,254 @@
+from main import get_timeseries_last_24h
+from main import apply_compare_mibs
+from main import WORKLOADS_LIST
+from main import HASH_TUPLE_LIST
+from unittest.mock import Mock
+import pandas as pd
+import numpy as np
+import time
+
+
+def __generate_datapoints(mean, std, api, samples_count):
+    # Generate samples for WRITE, READ[0], READ[1], READ[2]
+    # Predictable random numbers: we want to verify expected output and expected cases possible.
+    np.random.seed(0)
+    samples = np.random.normal(
+        mean, std, size=(samples_count * len(WORKLOADS_LIST) * len(HASH_TUPLE_LIST))
+    )
+    datapoint_list = []
+    idx = 0
+    for _ in range(0, samples_count):
+        for op in WORKLOADS_LIST:
+            for hash_tuple in HASH_TUPLE_LIST:
+                new_datapoint = {
+                    "ApiName": api,
+                    "Op": op,
+                    "Crc32cEnabled": hash_tuple[0],
+                    "MD5Enabled": hash_tuple[1],
+                    "MiBs": samples[idx],
+                }
+                idx += 1
+                datapoint_list.append(new_datapoint)
+    return datapoint_list
+
+
+def test_apply_compare_mibs():
+    data = __generate_datapoints(5, 2.5, "Xml", 1000)
+    data.extend(__generate_datapoints(5, 2.5, "Json", 1000))
+    test_df = pd.DataFrame.from_records(data)
+    results = apply_compare_mibs(test_df)
+    for op in WORKLOADS_LIST:
+        for hash_tuple in HASH_TUPLE_LIST:
+            assert results.loc[(op, hash_tuple[0], hash_tuple[1])] == False
+
+
+def test_convert_timeseries_to_dataframe():
+    series_single_datapoint = monitoring_v3.TimeSeries()
+    series_single_datapoint.metric.type = (
+        "custom.googleapis.com/cloudprober/external/workload_7/throughput"
+    )
+    series_single_datapoint.metric.labels["api"] = "Xml"
+    series_single_datapoint.metric.labels["op"] = "READ[0]"
+    series_single_datapoint.metric.labels["crc32c_enabled"] = "1"
+    series_single_datapoint.metric.labels["md5_enabled"] = "0"
+    series_single_datapoint.points = [
+        monitoring_v3.Point(
+            {"interval": current_interval(), "value": {"double_value": 10.0}}
+        )
+    ]
+    actual_df = convert_timeseries_to_dataframe([series_single_datapoint])
+    expected_df = pd.DataFrame.from_records(
+        [
+            {
+                "ApiName": "Xml",
+                "Op": "READ[0]",
+                "Crc32cEnabled": "1",
+                "MD5Enabled": "0",
+                "MiBs": 10.0,
+            }
+        ]
+    )
+    assert actual_df.equals(expected_df) == True
+
+
+def test_compare_mibs_null_hypothesis_distributions():
+    # Generate a random with different mean distributions
+    # That distributions are not the same
+    dataframe = __generate_datapoints(5, 2.5, "Xml", 1000)
+    dataframe.extend(__generate_datapoints(100, 2.5, "Json", 1000))
+    result = compare_mibs(pd.DataFrame.from_records(dataframe))
+    assert result == True
```

Comment:
Shouldn't the null hypothesis in this case be false as the distributions are not the same?

Reply:
The null hypothesis is "Are they different distributions?"; in this case it's True because they are the same. 
Updated comments for compare_mibs()

Reply:
Oh I see - is there a reason we flipped it around(`p < alpha` instead of `p > alpha`)? Usually Mann Whitney tests the probability of the same distribution so I got confused.

Reply:
The usual null for Mann-Whitney is a bit more complicated, but usually it is that they are the same:

https://en.wikipedia.org/wiki/Mann%E2%80%93Whitney_U_test

I am not a statistician, but I think usually the null is the thing that you would accept on weak or no evidence.  Given that the GCS team believes XML and JSON are the same, we just only change our minds (and their minds) on strong evidence.  In other words, we should stick to using "both are the same" for the null.

Just imagine a conversation where you can say: "You believe they are the same, but we have strong evidence that they are not."